backend/src/revisions/utils/extract-revision-metadata-from-content.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * SPDX-FileCopyrightText: 2023 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import {
+  convertRawFrontmatterToNoteFrontmatter,
+  defaultNoteFrontmatter,
+  extractFirstHeading,
+  extractFrontmatter,
+  generateNoteTitle,
+  NoteFrontmatter,
+  parseRawFrontmatterFromYaml,
+} from '@hedgedoc/commons';
+import { parseDocument } from 'htmlparser2';
+import MarkdownIt from 'markdown-it';
+
+/**
+ * Parses the frontmatter of the given content and extracts the metadata that are necessary to create a new revision..
+ *
+ * @param {string} content the revision content that contains the frontmatter.
+ */
+export function extractRevisionMetadataFromContent(content: string): {
+  title: string;
+  description: string;
+  tags: string[];
```

Comment:
Shouldn't this be declared as a type for consistency?

Reply:
What? The string array?